bosh-director/spec/unit/deployment_plan/instance_plan_spec.rb

```diff
@@ -359,6 +360,40 @@ module Bosh::Director::DeploymentPlan
           expect(instance_plan.needs_recreate?).to be_falsey
         end
       end
+
+      context 'when instance has unresponsive agent' do
+        let(:job_state) { 'unresponsive' }
+
+        it 'should return true' do
+          expect(instance_plan.needs_recreate?).to be_truthy
+        end
+      end
+    end
+
+    describe '#needs_to_fix?' do
+      context 'when instance has unresponsive agent' do
+        let(:job_state) { 'unresponsive' }
+
+        it 'should return true' do
+          expect(instance_plan.needs_to_fix?).to be_truthy
+        end
+      end
+
+      context 'when instance is ok' do
+        let(:instance_plan) { InstancePlan.new(existing_instance: existing_instance, desired_instance: desired_instance, instance: instance, network_plans: network_plans, recreate_deployment: true) }
+
+        it 'should return false' do
+          expect(instance_plan.needs_to_fix?).to be_falsey
+        end
+      end
+
+      context 'when instance is nill' do
+        let(:job_state) { 'running' }
```

Comment:
Is this the right `let` for this context? `job_state` returning a value of 'running' doesn't seem like it implies that the instance is nil.